torch/ao/quantization/backend_config/_pt2e.py

```diff
@@ -0,0 +1,151 @@
+    )
+    # TODO: remove when functionalization is supported in PT2 mode
+    conv_configs.append(
+        BackendPatternConfig((torch.ops.aten.relu_.default, torch.ops.aten.convolution.default))
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(dtype_configs)
+        ._set_input_type_to_index({"weight": 1, "bias": 2})
+    )
+    return conv_configs
+
+def get_pooling_configs():
+    backend_pattern_configs = []
+    observation_type = ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT
+    dtype_configs = [weighted_op_quint8_dtype_config]
+
+    def root_node_getter(node_pattern):
+        getitem, maxpool, index = node_pattern
+        return maxpool
+
+    backend_pattern_configs.append(
+        BackendPatternConfig((operator.getitem, torch.ops.aten.max_pool2d_with_indices.default, 0))
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(dtype_configs)
+        ._set_root_node_getter(root_node_getter)
+    )
+
+    return backend_pattern_configs
+
+def get_relu_configs():
+    backend_pattern_configs = []
+    observation_type = ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT
+    dtype_configs = [weighted_op_quint8_dtype_config]
+    backend_pattern_configs.append(
+        BackendPatternConfig(torch.ops.aten.relu.default)
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(dtype_configs))
+    return backend_pattern_configs
+
+def get_binary_op_configs():
+    binary_op_configs: List[BackendPatternConfig] = []
+    dtype_configs = [weighted_op_quint8_dtype_config]
+    num_tensor_args_to_observation_type_mapping = {
+        # TODO: this is not used right now since we have extra check in prepare
+        # will need to change this to NO_OBSERVER later after we implemented
+        # Tensor dtype inference properly
+        0: ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT,
+        1: ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT,
+        2: ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT,
+    }
+    for op_with_quantized_bop_scalar_variant in [torch.ops.aten.add.Tensor, torch.ops.aten.add_.Tensor]:
+        bop_patterns = [
+            # TODO: add the pattern
+            (torch.ops.aten.relu.default, op_with_quantized_bop_scalar_variant),
+            op_with_quantized_bop_scalar_variant,
+            # TODO: remove when functionalization is supported in pt2_mode
+            (torch.ops.aten.relu_.default, op_with_quantized_bop_scalar_variant),
+        ]
+        for bop_pattern in bop_patterns:
+            binary_op_configs.append(
+                BackendPatternConfig(bop_pattern)
+                    .set_dtype_configs(dtype_configs)  # noqa: E131
+                    ._set_num_tensor_args_to_observation_type(num_tensor_args_to_observation_type_mapping))
+
+    return binary_op_configs
+
+def get_pt2e_backend_config():
+    return BackendConfig("pytorch_2.0_export")  \
```

Comment:
PT2.0 export is not a backend in the way that backend_config is defined, should this be something like "qnnpack_pt2e" instead?

Reply:
hmmm, it's not pure qnnpack though, we may also have some quantized ops (e.g. quantized add, embedding_byte) that is run with executorch runtime

Reply:
it might not be pure qnnpack, but it's also definitely not a backend :)

If I understand the `backend_config` design correctly, each entry in a config should be for some hardware target + kernel library, for example "executorch runtime with QNNPACK".  Neither PT2.0 or executorch are tied to any single hardware target.

Can we make the name of the file and config more specific?

Reply:
This is exactly why we need some separation between qconfig and backend config. I am assuming that it is easier to say that give me a qconfig for this backend, but can we also have api that has some predefined qconfig names. Such as symmetric, asymmetric, etc. Would that be any better?

You would do backend related compatibility checks elsewhere. To be sure, I understand the appeal of backend config and I dont have any better ideas to make this cleaner, but I do see the confusion arising when we mix runtime with backend and delegate.

Reply:
@vkuzo I'm not sure what would be a good name for this, I'll rename it to _qnnpack_pt2e.py for now

@kimishpatel how is this related to qconfig? also not sure what is the confusion since we have explained what qconfig and backend_config means multiple times

Reply:
I am talking about the conversation we. had around relatiion of qconfig to backend config. Anyways, I dont want to drag this any further. Ok with whatever you guys decide.

Reply:
> @vkuzo I'm not sure what would be a good name for this, I'll rename it to _qnnpack_pt2e.py for now

I think "qnnpack_pt2e" is actually the correct name.  In the test case, we are comparing the results of this workflow with the FX graph mode quantization workflow with the qnnpack backend_config and expecting them to match.

Reply:
sounds good, my main point about this is that we also have some operators that is not run with "qnnpack" the library (e.g. embedding_byte), but "qnnpack" as a backend in pytorch ao means something different, it can contain these ops as well. it might be helpful to clarify these things as well I think

Reply:
that makes sense.  I think it's the same story with non-executorch usage of qnnpack, right?

I guess I'd prefer:
1. names are consistent with each other (if we have backend_config_qnnpack, the pt2e version should be backend_config_qnnpack_pt2e)
2. names make as much sense as possible (maybe we should rename backend_config_qnnpack to something more specific)

I was just commenting on (1), but (2) can also be a future improvement to all backend configs

Reply:
yeah agree, I think we just need to document what these means exactly